src/test/kotlin/org/rust/ide/inspections/lints/RsNeedlessLifetimesInspectionTest.kt

```diff
@@ -355,6 +355,14 @@ class RsNeedlessLifetimesInspectionTest : RsInspectionsTestBase(RsNeedlessLifeti
         fn foo2<'a>(_: &'a str) {}
     """)
 
+    fun `test async fn`() = checkByText("""
```

Comment:
I am afraid this test is broken (as well as the previous `test allow` test) because `checkByText` does not check weak warnings by default.
I have already mentioned this problem in another PR https://github.com/intellij-rust/intellij-rust/pull/9076#discussion_r949007967 and I think we should really do something to improve this API for the future 🙂